### PR TITLE
LibWeb: Implement `document.createCDATASection()`

### PR DESCRIPTION
**Pull request by tcl3**
This PR adds `document.createCDATASection()` and includes some fixes to XML document serialization and deserialization, which were necessary to get the `createCDATASection` test to pass.

This method is used quite extensively by the WPT test harmess, so having it should allow us to run quite a few more WPT tests.

Before:
create_cdata_section_before

After:
create_cdata_section_after

### CODE REVIEW

File: Userland/Libraries/LibWeb/DOM/Document.h [hunk 1]
```diff
@@ -642,7 +642,7 @@ class Document
     QuirksMode m_quirks_mode { QuirksMode::No };
 
     // https://dom.spec.whatwg.org/#concept-document-type
-    Type m_type { Type::HTML };
+    Type m_type { Type::XML };
```
Comment (gmta) [inline Userland/Libraries/LibWeb/DOM/Document.h:646]: Please state in the commit message _why_ you're changing this, not _what_ you're changing (which is already obvious from the diff).

Comment (tcl3) [reply Userland/Libraries/LibWeb/DOM/Document.h:646]: I've updated the commit message to better explain why I made the change.

tl;dr: because the spec says so.

**Comment by tcl3** [comment]
FWIW, I ran all of the WPT DOM tests that use testharness.js with the following command:

`/home/tim/repos/serenity/Tests/LibWeb/WPT/wpt/wpt run --webdriver-binary /home/tim/repos/serenity/Build/lagom/bin/WebDriver --log-raw wpt.log ladybird dom --test-types testharness`

I got these results:

Before:
Ran 489 tests finished in 2620.4 seconds.
  • 203 ran as expected. 0 tests skipped.
  • 14 tests crashed unexpectedly
  • 56 tests had errors unexpectedly
  • 66 tests timed out unexpectedly
  • 165 tests had unexpected subtest results
  
  
After:
Ran 489 tests finished in 2601.8 seconds.
  • 208 ran as expected. 0 tests skipped.
  • 14 tests crashed unexpectedly
  • 56 tests had errors unexpectedly
  • 65 tests timed out unexpectedly
  • 160 tests had unexpected subtest results

**Review by gmta — APPROVED** [review]

**Comment by awesomekling** [comment]
Excellent!
